### ISSUE TITLE
Program mode

### ISSUE THREAD

**Issue by bragitman** [cyr-ius/hass-heatzy#25]
Does hass-heatzy support Program Mode?  When I set Heatzy Pilote or Heatzy Glow into Program Mode (using the Heatzy App) I cannot see this shown in Home Assistant.  Also, in Home Assistant I cannot see anyway to start or stop Program Mode .  Thanks for your great work.

**Comment by Cyr-ius** [comment]
> Does hass-heatzy support Program Mode? When I set Heatzy Pilote or Heatzy Glow into Program Mode (using the Heatzy App) I cannot see this shown in Home Assistant. Also, in Home Assistant I cannot see anyway to start or stop Program Mode . Thanks for your great work.

What are the elements that you would like to see appear because in general we use HA to replace the program mode?
there is a status that shows if the module is in Programmed mode or not, (I could show you this one), or even make you a service that will enable it to be activated if necessary?

**Comment by bragitman** [comment]
Bonjour.

I have Heatzy Glow and Heatzy Pilote in a holiday rental property.   After each guest leaves I want to use HA to automatically reset the settings - so set the eco/cmft temperatures and ensure they're in program mode.  I also want to use HA to view the history of any changes the guests make.   

This week I have modified your 5.7.0 code to be able to do the following:

1. Sync Program/Timer Mode
2. Heatzy Vacation mode is now shown in HA as AWAY mode.  It's not possible to set Vacation mode on HA.
3. HA Target Temperature added, this is based on the current Cmft/Eco/Fro mode and the temperatures for them.  Any change made to HA Target Temperature is not sync'd to Heatzy.
4. HVAC Action added.  For PILOTE this is either OFF/HEATING, based on the preset mode.  For GLOW this is OFF/HEATING/IDLE, only set to HEATING if the Target Temperature is above the Current Temperature.

I can share the code with you - is the best way for me to create a Github branch with my changes?

**Comment by Cyr-ius** [comment]
> Bonjour.
> 
> I have Heatzy Glow and Heatzy Pilote in a holiday rental property. After each guest leaves I want to use HA to automatically reset the settings - so set the eco/cmft temperatures and ensure they're in program mode. I also want to use HA to view the history of any changes the guests make.
> 
> This week I have modified your 5.7.0 code to be able to do the following:
> 
>     1. Sync Program/Timer Mode
> 
>     2. Heatzy Vacation mode is now shown in HA as AWAY mode.  It's not possible to set Vacation mode on HA.
> 
>     3. HA Target Temperature added, this is based on the current Cmft/Eco/Fro mode and the temperatures for them.  Any change made to HA Target Temperature is not sync'd to Heatzy.
> 
>     4. HVAC Action added.  For PILOTE this is either OFF/HEATING, based on the preset mode.  For GLOW this is OFF/HEATING/IDLE, only set to HEATING if the Target Temperature is above the Current Temperature.
> 
> 
> I can share the code with you - is the best way for me to create a Github branch with my changes?

I made a version that uses your idea to reset override modes when handling the component. I also removed the PRESET_NONE mode which is useless
I would be interested in your opinion, especially since I cannot test the GLOW, not having this one
I am French so if the language of Molière inspires you ;-)

this version is available in the branch: with_derog

https://github.com/Cyr-ius/hass-heatzy/tree/with_derog/custom_components/heatzy

**Comment by bragitman** [comment]
Thanks, I like the removal of PRESET_NONE.

I needed to update the const.py file with the new constants.  And I needed to fix the imports.

I have also made a few changes for PiloteV2 as this wasn't working correctly for me.  Updating preset mode cannot be made in the same call as turning off the vacation or program mode.

And for GlowV2 I removed the setting of the preset mode when turning on.  Setting preset mode when turning on made sense when PRESET_NONE was used, but not now.

Here is the code with my updates:
[https://github.com/bragitman/hass-heatzy/tree/with_derog/custom_components/heatzy](url)
